ui/app/helpers/utils/conversion-util.js

```diff
@@ -60,60 +58,75 @@ const baseChange = {
-  R.not,
-  R.eqBy(R.__, 'fromCurrency', 'toCurrency'),
-  R.flip(R.prop)
-)
-
-// Lens
-const valuePropertyLens = R.over(R.lensProp('value'))
-const conversionRateLens = R.over(R.lensProp('conversionRate'))
-
-// conditional conversionRate setting wrapper
-const whenPredSetCRWithPropAndSetter = (pred, prop, setter) => R.when(
-  pred,
-  R.converge(
-    conversionRateLens,
-    [R.pipe(R.prop(prop), setter), R.identity]
-  )
-)
-
-// conditional 'value' setting wrappers
-const whenPredSetWithPropAndSetter = (pred, prop, setter) => R.when(
-  pred,
-  R.converge(
-    valuePropertyLens,
-    [R.pipe(R.prop(prop), setter), R.identity]
-  )
-)
-const whenPropApplySetterMap = (prop, setterMap) => whenPredSetWithPropAndSetter(
-  R.prop(prop),
-  prop,
-  R.prop(R.__, setterMap)
-)
-
-// Conversion utility function
-const converter = R.pipe(
-  whenPredSetCRWithPropAndSetter(R.prop('conversionRate'), 'conversionRate', decToBigNumberViaString),
-  whenPredSetCRWithPropAndSetter(R.prop('invertConversionRate'), 'conversionRate', invertConversionRate),
-  whenPropApplySetterMap('fromNumericBase', toBigNumber),
-  whenPropApplySetterMap('fromDenomination', toNormalizedDenomination),
-  whenPredSetWithPropAndSetter(fromAndToCurrencyPropsNotEqual, 'conversionRate', convert),
-  whenPropApplySetterMap('toDenomination', toSpecifiedDenomination),
-  whenPredSetWithPropAndSetter(R.prop('numberOfDecimals'), 'numberOfDecimals', round),
-  whenPredSetWithPropAndSetter(R.prop('roundDown'), 'roundDown', roundDown),
-  whenPropApplySetterMap('toNumericBase', baseChange),
-  R.view(R.lensProp('value'))
-)
+/**
+ * Defines the base type of numeric value
+ * @typedef {('hex' | 'dec' | 'BN')} NumericBase
+ */
+
+/**
+ * Defines which type of denomination a value is in
+ * @typedef {('WEI' | 'GWEI' | 'ETH')} EthDenomination
+ */
+
+/**
+ * Utility method to convert a value between denominations, formats and currencies.
+ * @param {Object} input
+ * @param {string | BigNumber} input.value
+ * @param {NumericBase} input.fromNumericBase
+ * @param {EthDenomination} [input.fromDenomination]
+ * @param {string} [input.fromCurrency]
+ * @param {NumericBase} input.toNumericBase
+ * @param {EthDenomination} [input.toDenomination]
+ * @param {string} [input.toCurrency]
+ * @param {number} [input.numberOfDecimals]
+ * @param {number} [input.conversionRate]
+ * @param {boolean} [input.invertConversionRate]
+ * @param {string} [input.roundDown]
+ */
+const converter = ({
+  value,
+  fromNumericBase,
+  fromDenomination,
+  fromCurrency,
+  toNumericBase,
+  toDenomination,
+  toCurrency,
+  numberOfDecimals,
+  conversionRate,
+  invertConversionRate,
+  roundDown,
+}) => {
+  let convertedValue = toBigNumber[fromNumericBase ?? 'dec'](value)
+
+  if (fromDenomination) {
+    convertedValue = toNormalizedDenomination[fromDenomination](convertedValue)
+  }
+
+  if (fromCurrency !== toCurrency) {
+    let rate = toBigNumber.dec(conversionRate ?? 1)
```

Comment:
is it preferable for it to blow up?

Reply:
I can see no advantage letting it blow up here though 🤔 🤷‍♂️

Reply:
the old version checked if fromCurrency didn't match toCurrency and then expected conversionRate to be supplied... I thought about throwing an error in the case it wasn't because... otherwise it isn't doing what the user intended.

Reply:
Oh, well, I guess it's a good thing to crash if the conversion rate is missing but a conversion is being explicitly asked for. Better to blow up than show the wrong answer.